plugins/opds.koplugin/opdsbrowser.lua

```diff
@@ -695,7 +695,47 @@ function OPDSBrowser:downloadFile(item, filename, remote_url)
     end
 end
 
-function OPDSBrowser:streamPages(item, remote_url, count)
+-- This function shows a dialog with input fields
+-- for entering information for Page streaming from.
+-- an input page number
+function OPDSBrowser:jumpToPage(viewer, count)
+    local input_dialog
+    input_dialog = InputDialog:new{
+        title = "Enter Page Number",
+        input = "",
+        input_type = "number",
+        input_hint = "Page # (1 - "..count..")",
+        buttons = {
+            {
+                {
+                    text = _("Cancel"),
+                    id = "close",
+                    callback = function()
+                        UIManager:close(input_dialog)
+                    end,
+                },
+                {
+                    text = _("Stream"),
+                    is_enter_default = true,
+                    callback = function()
+                        UIManager:close(input_dialog)
+                        if (input_dialog:getInputValue() > count) then
```

Comment:
You will get a crash on entering a string that cannot be converted to number.

Reply:
Added a check to make sure input is not nil. If it is nil, an error message pops up.

Reply:
We usually just do not accept incorrect input, so it can be smth like
```Lua
local page_num = input_dialog:getInputValue()
if page_num then
    UIManager:close(input_dialog)
    viewer:switchToImageNum(math.min(math.max(1, page_num), count))
end
```

Reply:
Now that's neat! I haven't seen that before, I'll implement it in the next commit.